src/main/java/com/faforever/client/main/MainController.java

```diff
@@ -338,6 +339,20 @@ public void display() {
     registerWindowListeners();
   }
 
+  private void hideSplashScreen() {
+    try {
+      Class splashScreenClass = Class.forName("com.install4j.api.launcher.SplashScreen");
+      Method hideMethod = splashScreenClass.getDeclaredMethod("hide");
+      hideMethod.invoke(null);
+    } catch (ClassNotFoundException e) {
+      log.debug("No install4j splash screen found to close.");
+    } catch (NoSuchMethodException | IllegalAccessException e) {
```

Comment:
This would be unexpected, so I suggest to log an error. Same for the InvocationTargetException which means that the called method has thrown an error (like ConnectionException of hide()).

Reply:
So you mean error instead of warning?

Reply:
Yes.